### PR TITLE
Fix accounts-v2 deposit and related bugs

### LINKED ISSUE #7164: Error when trying to auto deposit using accounts-v2 deposit
# 🐞 Bug Report

### Description

Receiving the following error when running accounts-v2 deposit:

```[2020-09-02 11:40:04] FATAL accounts-v2: Could not send validator deposit(s): read /home/myuser/prysm: is a directory```

### Has this worked before in a previous version?

N/A - not sure

## 🔬 Minimal Reproduction

Used the following command to invoke:
```./prysm.sh validator accounts-v2 deposit --wallet-dir /home/myuser/.eth2validators/whale_prysm --eth1-keystore-utc-path /home/myuser/.ethereum/goerli/keystore --eth1-keystore-password-file /home/myuser/mypasswdfile --deposit-all-accounts --http-web3provider http://localhost:8546```

## 🔥 Error

<pre><code>
```[2020-09-02 11:40:04] FATAL accounts-v2: Could not send validator deposit(s): read /home/myuser/prysm: is a directory```
</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
    Ubuntu 18.04
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
    validator-v1.0.0-alpha.24-linux-amd64
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

Initial output confirming my flags:
```Primary key fingerprint: 0AE0 051D 647B A3C1 A917  AF40 72E3 3E4D F1A5 036E
Verified /home/myuser/prysm/dist/validator-v1.0.0-alpha.24-linux-amd64 has been signed by Prysmatic Labs.
Starting Prysm validator accounts-v2 deposit --wallet-dir /home/myuser/.eth2validators/whale_prysm --eth1-keystore-utc-path /home/myuser/.ethereum/goerli/keystore --eth1-keystore-password-file /home/myuser/mypasswdfile --deposit-all-accounts --http-web3provider http://localhost:8546
[2020-09-02 12:15:32]  INFO accounts-v2: (wallet directory) /home/myuser/.eth2validators/whale_prysm```

### PR DESCRIPTION
**Pull request by dv8silencer**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

Issue #7164 outlines a bug with depositing using cli:  
(from @danielschonfeld )

`./prysm.sh validator accounts-v2 deposit --wallet-dir /home/myuser/.eth2validators/whale_prysm --eth1-keystore-utc-path /home/myuser/.ethereum/goerli/keystore --eth1-keystore-password-file /home/myuser/mypasswdfile --deposit-all-accounts --http-web3provider http://localhost:8546
`

Creating this output 

```[2020-09-02 11:40:04] FATAL accounts-v2: Could not send validator deposit(s): read /home/myuser/prysm: is a directory```

In the path to resolving this bug, its cause and other bugs were discovered and are addressed by this PR:

- validator/accounts/v2/accounts_deposit.go line 76:  Based on the already written-comment in code ("If the user has selected to deposit all accounts, we skip this part.") and review of the logic, the wrong flag was being checked in the if; it should be DepositAllAccountsFlag
- validator/accounts/v2/accounts_deposit.go line 133:  This logic is meant to capture cases of `hasPrivateKey || hasEth1Keystore` but only deals with what should happen if Eth1PrivateKeyFileFlag has been provided (hasPrivateKey).  So if a user submits a path to an Eth1Keystore, this part of the logic is hit and soon returns from the function, but that flag and its content is never regarded.
- validator/accounts/v2/prompt.go line 64:  The codepaths that lead to the function call inputWeakPassword() expect it return a path, not read the content into the return variable.  While testing this, you can craft the command line arguments so that the program ends up trying to look for a file such as /path/to/a/path/(contentOfFile) because the function returns the content of the file, not the path.  I changed it so it returns the path.

**Which issues(s) does this PR fix?**

Fixes #7164

### CODE REVIEW

File: validator/accounts/v2/accounts_deposit.go [hunk 1]
```diff
@@ -137,6 +137,20 @@ func createDepositConfig(cliCtx *cli.Context, km *derived.Keymanager) (*derived.
 				return nil, err
 			}
 			config.Eth1PrivateKey = strings.TrimRight(string(fileBytes), "\r\n")
+		} else {
+			config.Eth1KeystoreUTCFile = cliCtx.String(flags.Eth1KeystoreUTCPathFlag.Name)
```
Comment (terencechain) [inline validator/accounts/v2/accounts_deposit.go:141]: Can we please test this logic? Thanks!

**Review by terencechain — CHANGES_REQUESTED** [review]

File: validator/accounts/v2/accounts_deposit.go (hunk 1 above)
Comment (dv8silencer) [reply validator/accounts/v2/accounts_deposit.go:141]: Working on it.  Thanks

Comment (dv8silencer) [reply validator/accounts/v2/accounts_deposit.go:141]: Addressed.  Can you take a look at the tests?

File: validator/accounts/v2/accounts_deposit_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	eth1PrivateKeyFile          string
+	httpWeb3ProviderFlag        string
+	publicKeysFlag              string
+	depositAllAccountsFlag      bool
+	skipDepositConfirmationFlag bool
+	keymanagerKind              v2keymanager.Kind
+}
+
+func setupWalletCtxforDeposits(
+	tb testing.TB,
```
Comment (terencechain) [inline validator/accounts/v2/accounts_deposit_test.go:36]:
```suggestion
	t testing.T,
```

Don't think we need `b` here. (it's used for benchmark)

File: validator/accounts/v2/accounts_deposit_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+		keymanagerKind:              v2keymanager.Derived,
+		walletDir:                   walletDir,
+		walletPasswordFile:          passwordFilePath,
+		skipDepositConfirmationFlag: true,
+		depositAllAccountsFlag:      true,
+		httpWeb3ProviderFlag:        "http://localhost:8545",
+		eth1PrivateKeyFile:          eth1PrivateKeyFile.Name(),
+	})
+	require.NoError(t, err)
+	if len(depositConfig.DepositPublicKeys) != 3 {
```
Comment (terencechain) [inline validator/accounts/v2/accounts_deposit_test.go:132]:
```suggestion
	require.Equal(t, 3, len(depositConfig.DepositPublicKeys), "wrong number of public keys")
```

Same with the occurrences below

File: validator/accounts/v2/accounts_deposit_test.go (hunk 2 above)
Comment (dv8silencer) [reply validator/accounts/v2/accounts_deposit_test.go:36]: Will address in a commit soon.  Thanks!

File: validator/accounts/v2/accounts_deposit_test.go (hunk 3 above)
Comment (dv8silencer) [reply validator/accounts/v2/accounts_deposit_test.go:132]: Will address in a commit soon. Thanks!

File: validator/accounts/v2/accounts_deposit_test.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+		skipDepositConfirmationFlag: true,
+		depositAllAccountsFlag:      true,
+		httpWeb3ProviderFlag:        "http://localhost:8545",
+		eth1PrivateKeyFile:          eth1PrivateKeyFile.Name(),
+	})
+	require.NoError(t, err)
+
+	require.Equal(t, 3, len(depositConfig.DepositPublicKeys), "wrong number of public keys")
+
+	if depositConfig.Eth1PrivateKey != "This should be an ETH1 private key" {
```
Comment (terencechain) [inline validator/accounts/v2/accounts_deposit_test.go:135]: `require.Equal(t, "This should be an ETH1 private key", depositConfig.Eth1PrivateKey, "eth1 private key does not match")`

can do the same thing here + a few places below : )

Comment (dv8silencer) [reply validator/accounts/v2/accounts_deposit_test.go]: Done!

File: validator/accounts/v2/accounts_deposit_test.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	require.Equal(t, 3, len(depositConfig.DepositPublicKeys), "wrong number of public keys")
+	require.Equal(t, "This would be eth1 keystore file path", depositConfig.Eth1KeystoreUTCFile,
+		"eth1 keystore file path incorrect")
+	require.Equal(t, "This would be eth1 keystore password file path", depositConfig.Eth1KeystorePasswordFile,
+		"eth1 keystore password file path incorrect")
+	require.Equal(t, "", depositConfig.Eth1PrivateKey, "eth1 private key should be empty string")
+}
+
+// createDepositConfigHelper returns a SendDepositConfig when given a particular wallet configuration.
+func createDepositConfigHelper(t *testing.T, config *depositTestWalletConfig) (*derived.SendDepositConfig, error) {
```
Comment (terencechain) [inline validator/accounts/v2/accounts_deposit_test.go:183]:
```suggestion
func createDepositConfigHelper(t *testing.T, config *depositTestWalletConfig) *derived.SendDepositConfig {
```

Don't need to return the error if we were doing to pass in `t` and assert error within this function

File: validator/accounts/v2/accounts_deposit_test.go [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+	})
+	require.NoError(t, err)
+
+	keymanager, err := wallet.InitializeKeymanager(
+		cliCtx.Context,
+		true, /* skip mnemonic confirm */
+	)
+	require.NoError(t, err)
+	km, ok := keymanager.(*derived.Keymanager)
+	if !ok {
```
Comment (terencechain) [inline validator/accounts/v2/accounts_deposit_test.go:198]:
```suggestion
require.Equal(t, true, ok, "keymanager must be derived type")
```

Comment (dv8silencer) [reply validator/accounts/v2/accounts_deposit_test.go:198]: Thanks.  Will address in next commit soon.

File: validator/accounts/v2/accounts_deposit_test.go (hunk 5 above)
Comment (dv8silencer) [reply validator/accounts/v2/accounts_deposit_test.go:183]: Thanks.  Will address in next commit soon.

**Review by terencechain — APPROVED** [review]

**Pull request merged**